test/lsp.py

```diff
@@ -898,18 +898,28 @@ def main(self) -> int:
 
         return min(max(self.test_counter.failed, self.assertion_counter.failed), 127)
 
-    def setup_lsp(self, lsp: JsonRpcProcess, expose_project_root=True):
+    def setup_lsp(
+        self,
+        lsp: JsonRpcProcess,
+        expose_project_root=True,
+        analyze_all_files_in_project=False,
+        project_root_subdir=None
+    ):
         """
         Prepares the solc LSP server by calling `initialize`,
         and `initialized` methods.
         """
+        project_root_dir_uri_with_maybe_subdir = self.project_root_uri
```

Comment:
You can actually shorten that a little bit:
```suggestion
        project_root_uri_with_maybe_subdir = self.project_root_uri
```

Also, another possibility:
```suggestion
        project_root_uri_override = self.project_root_uri
```
Not great but at least it makes it clear why you have two similarly named things (you want to override the original value) and which you should use (the override unless you're interested in the original value for some reason).